### PR TITLE
Add note for HCE application performance

### PR DESCRIPTION
**Pull request by jh8801jung**
### Change Description ###

Add note for HCE application performance

### CODE REVIEW

**Comment by huytaduc** [comment]
checkout the webpage https://docs.stg.tizen.org/staging/616

File: docs/application/native/guides/connectivity/nfc.md [hunk 1]
```diff
@@ -98,6 +98,13 @@ The main features of the NFC API include:
 
   ![HCE routing](./media/nfc_hce_routing.png)
 
+  > **Note**
+  >
+  > Pay attention to the following:
```
Comment (nawab-samsung) [inline docs/application/native/guides/connectivity/nfc.md]: @jh8801jung 

Global comment: 

- Remove the instances of sentence **Pay attention to the following:** in whole document, as we are already giving a **Note**.
- Note alone serves the said purpose.

File: docs/application/native/guides/connectivity/nfc.md [hunk 2]
```diff
@@ -98,6 +98,13 @@ The main features of the NFC API include:
...
+  > - HCE operation requires launching HCE application in Tizen Application Framework. Because of this, enough time is required for the data to be transferred to the application.
```
Comment (nawab-samsung) [inline docs/application/native/guides/connectivity/nfc.md]: HCE operations begin with the launch of HCE application in Tizen Application Framework. With the execution of these operations, the data transfer to the application needs more time.

File: docs/application/native/guides/connectivity/nfc.md [hunk 3]
```diff
@@ -98,6 +98,13 @@ The main features of the NFC API include:
...
+  > - If you consider time efficient, we recommend that you don't create HCE applications and communicate with the Secure Elements provided in the device.
```
Comment (nawab-samsung) [inline docs/application/native/guides/connectivity/nfc.md]: >- In case, you consider time efficiency, it is recommend not to create HCE applications and communicate with the Secure Elements provided in the device.

File: docs/application/native/guides/connectivity/nfc.md [hunk 4]
```diff
@@ -98,6 +98,13 @@ The main features of the NFC API include:
...
+  > - Alternatively, make sure that the HCE application is always in the launching state.
```
Comment (nawab-samsung) [inline docs/application/native/guides/connectivity/nfc.md]: >- Alternatively, ensure that the HCE application is always in the launch state.

File: docs/application/native/guides/connectivity/nfc.md [hunk 5]
```diff
@@ -98,6 +98,13 @@ The main features of the NFC API include:
...
+
 ## Prerequisites
 
 To enable your application to use the NFC functionality:
```
Comment (nawab-samsung) [inline docs/application/native/guides/connectivity/nfc.md]: To enable use of NFC functionality in your application:

File: docs/application/native/guides/connectivity/nfc.md [hunk 6]
```diff
@@ -98,9 +98,15 @@ The main features of the NFC API include:
 
   ![HCE routing](./media/nfc_hce_routing.png)
 
+  > **Note**
+  >
+  > - HCE operations begin with the launch of HCE application in Tizen Application Framework. With the execution of these operations, the data transfer to the application needs more time.
+  > - In case, you consider time efficiency, it is recommend not to create HCE applications and communicate with the Secure Elements provided in the device.
```
Comment (mijin85cho) [inline docs/application/native/guides/connectivity/nfc.md]: @nawab-samsung How about keeping "If" instead of "In case,"?

If you consider time efficiency, ~

File: docs/application/native/guides/connectivity/nfc.md [hunk 7]
```diff
@@ -98,9 +98,15 @@ The main features of the NFC API include:
...
+  > - Alternatively, ensure that the HCE application is always in the launch state.
```
Comment (mijin85cho) [inline docs/application/native/guides/connectivity/nfc.md]: I'm not sure with "launch state"..

Hello, @annie-samsung.
Would you please check the "launch state" in this sentence?

File: docs/application/native/guides/connectivity/nfc.md (hunk 6 above)
Comment (nawab-samsung) [reply docs/application/native/guides/connectivity/nfc.md]: @mijin85cho 
I wrote "incase" because there might be cases where we think about time efficiency ,else we do not consider the time efficiency.

The thing is when we think of time efficiency , that is a certain condition/criteria associated as: "when" "why"

Comment (mijin85cho) [reply docs/application/native/guides/connectivity/nfc.md]: @nawab-samsung I'm suggesting "if" because of the same reason with your comment.
"if you consider the time efficiency" goes with "else, you don't consider". 

With "In case, you consider time efficiency, ", I thought there are too many commas than required, stopping the flow of the sentence without meaning. Of course the condition is important, but more simple word would be better.

File: docs/application/native/guides/connectivity/nfc.md [hunk 8]
```diff
@@ -100,14 +100,13 @@ The main features of the NFC API include:
...
-  > - If you consider time efficient, we recommend that you don't create HCE applications and communicate with the Secure Elements provided in the device.
-  > - Alternatively, make sure that the HCE application is always in the launching state.
+  > - HCE operations begin with the launch of HCE application in Tizen Application Framework. With the execution of these operations, the data transfer to the application needs more time.
+  > - In case, you consider time efficiency, it is recommend not to create HCE applications and communicate with the Secure Elements provided in the device.
+  > - Alternatively, ensure that the HCE application is always in the launch state.
 
 ## Prerequisites
 
-To enable your application to use the NFC functionality:
+To enable use of NFC functionality in your application:
```
Comment (nawab-samsung) [inline docs/application/native/guides/connectivity/nfc.md]: - Enable use of NFC functionality in your application:

Comment (mijin85cho) [reply docs/application/native/guides/connectivity/nfc.md]: Hello, @nawab-samsung 
This "Prerequisites" section is used to give the **precondition** to use the function.
If you check other pages also, you'd see they also have "Prerequisites" section to describe the things what to do **before** you try to implement the function.

Your suggestion means a user would enable the NFC function by below steps,
so I don't thing it is correct sentence in here.

Also, as we have lots of guide pages,
please check the consistency with other pages.
We have several members in SRI-B reviewer team, and each member tries to change the sentence as their styles.
It is causing various  styles in the document format, and I think following one fixed style is better than that.

**Comment by jh8801jung** [comment]
> @jh8801jung
> Kindly do the required changes, such that I can verify and approve.

Finally, can you summarize the sentences? It seems that the comment is divided, but I can not see your comment on it.

**Comment by mijin85cho** [comment]
> ======================<<< **WARNING: Coding Style** >>>=======================
> docs/application/native/guides/connectivity/nfc.md: 114: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:ExV)
> docs/application/native/guides/connectivity/nfc.md: 114: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxE)
> docs/application/native/guides/connectivity/nfc.md: 115: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:ExV)
> docs/application/native/guides/connectivity/nfc.md: 115: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxV)
> docs/application/native/guides/connectivity/nfc.md: 115: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxE)
> docs/application/native/guides/connectivity/nfc.md: 116: ERROR: [SPC_M_SEP] space prohibited between function name and open parenthesis '('
> docs/application/native/guides/connectivity/nfc.md: 116: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:ExO)
> docs/application/native/guides/connectivity/nfc.md: 116: ERROR: [SPC_M_OPR] space required before that '!' (ctx:OxO)
> ...


@jh8801jung
Please ignore this message. We are setting Coding Style Checker, and it doesn't distinguish xml code yet.

File: docs/application/native/guides/connectivity/nfc.md (hunk 8 above)
Comment (nawab-samsung) [reply docs/application/native/guides/connectivity/nfc.md]: @mijin85cho  
Thanks for the clarity, I do understand that "to do X one must ensure that certain activities are done.
I have written the sentence in the context that Enabling NFC is one of the prerequisites.

Comment (nawab-samsung) [reply docs/application/native/guides/connectivity/nfc.md]: # Prerequisites

- You must enable NFC functionality in your application.

File: docs/application/native/guides/connectivity/nfc.md [hunk 9]
```diff
@@ -98,9 +98,15 @@ The main features of the NFC API include:
...
+  > - HCE operations begin with the launch of HCE application in Tizen Application Framework. With the execution of these operations, the data transfer to the application needs more time.
+  > - In case, you consider time efficiency, it is recommend not to create HCE applications and communicate with the Secure Elements provided in the device.
+  > - Alternatively, ensure that the HCE application is always in the launch state.
+
 ## Prerequisites
 
-To enable your application to use the NFC functionality:
+To enable use of NFC functionality in your application:
 
 1. To use the NFC API (in [mobile](../../api/mobile/latest/group__CAPI__NETWORK__NFC__MODULE.html) and [wearable](../../api/wearable/latest/group__CAPI__NETWORK__NFC__MODULE.html) applications), the application has to request permission by adding the following privileges to the `tizen-manifest.xml` file:
```
Comment (nawab-samsung) [inline docs/application/native/guides/connectivity/nfc.md]: @jh8801jung
Kindly incorporate the following: 

**Action item 1:** 
**Global Comment** 
Change all numbered lists in the document as shown in the example:  
1. -> 1) 
2.-> 2)
3.-> 3) 
4.-> 4) 
.
.
10.-> 10)

**Action item 2:** 
**Global Comment** 
Remove the following line just below the diagram after **Note**

Pay attention to the following:

**Action item 3:**
 
Replace 
1. To use the NFC API (in [mobile](../../api/mobile/latest/group__CAPI__NETWORK__NFC__MODULE.html) and [wearable](../../api/wearable/latest/group__CAPI__NETWORK__NFC__MODULE.html) applications), the application has to request permission by adding the following privileges to the `tizen-manifest.xml` file:

with: 

1) To use the NFC API in ([mobile](../../api/mobile/latest/group__CAPI__NETWORK__NFC__MODULE.html) and [wearable](../../api/wearable/latest/group__CAPI__NETWORK__NFC__MODULE.html) applications), the application has to request permission by adding the following privileges to the `tizen-manifest.xml` file:

Comment (nawab-samsung) [reply docs/application/native/guides/connectivity/nfc.md]: @mijin85cho  now the whole prerequisites looks like this 

# Prerequisites

You must enable NFC functionality in your application and ensure:

1) To use the NFC API in ([mobile](../../api/mobile/latest/group__CAPI__NETWORK__NFC__MODULE.html) and [wearable](../../api/wearable/latest/group__CAPI__NETWORK__NFC__MODULE.html) applications), the application has to request permission by adding the following privileges to the `tizen-manifest.xml` file:

File: docs/application/native/guides/connectivity/nfc.md (hunk 7 above)
Comment (nawab-samsung) [reply docs/application/native/guides/connectivity/nfc.md]: @jh8801jung 
This sentence is little ambiguous and both @mijin85cho and myself are not satisfied with it. 

I am giving you certain alternatives, if those make sense, kindly replace then: 

**Original Sentence was:** 
 Alternatively, ensure that the HCE application is always in the launching state.

I had changed it to: 
Alternatively, ensure that the HCE application is always in the launch state.

Suggested alternative: 
Alternatively, ensure that the HCE application is always in the ready state.

@mijin85cho  viewpoint was: 

Alternatively, ensure that the HCE application is always in the running state.

File: docs/application/native/guides/connectivity/nfc.md (hunk 8 above)
Comment (mijin85cho) [reply docs/application/native/guides/connectivity/nfc.md]: > I have written the sentence in the context that Enabling NFC is one of the prerequisites.

@nawab-samsung Your newly suggested sentence is totally reversing the order.
The list items  in this section is not based on "Enabling NFC is one of the prerequisites".
Please check that this section is "Prerequisites", which means these **should be prepared to enable the NFC**.
To enable the NFC, actions described in this section must be set, otherwise you **can't** enable the NFC. 

To enable the NFC, 
   1. you must add the `http://tizen.org/privilege/nfc` privileges to the `tizen-manifest.xml`
   2. you must add `<nfc.h>` header
   3. Support to the NFC should be checked.... and so on.

It is not the "enable the NFC functionality and ensure."
"To enable NFC, ensure" is the meaning of this section.

File: docs/application/native/guides/connectivity/nfc.md (hunk 9 above)
Comment (mijin85cho) [reply docs/application/native/guides/connectivity/nfc.md]: Question to @nawab-samsung 

>Action item 1:
Global Comment
Change all numbered lists in the document as shown in the example:

What is the reason that you decided to break the style rule of the list?
https://github.com/Samsung/tizen-docs/blob/master/styleguide/style.md#lists


>Action item 3:

What is the reason of moving the location of "("?
The original sentence means
"To use the NFC API in mobile and in wearable applications, " 
The "(in mobile and wearable applications)" is used as both mobile/wearable profiles are supporting NFC. We have 2 profiles for native NFC, so 2 API links are needed. One to mobile native, and one to wearable native. 
To add 2 links to here, as you can see "(in mobile and wearable applications)" is used. If only one profile supported the NFC, the link to API reference would have linked to "To use the NFC API," and no additional parenthesis would have added.

Please refer [this page](https://github.com/Samsung/tizen-docs/blob/master/docs/application/native/guides/connectivity/mtp.md), the example of only one profile (mobile in this case) support case.
image

Other [example page](https://github.com/Samsung/tizen-docs/blob/master/docs/application/native/guides/messaging/messages.md), showing the case of mobile/wearable profiles are supported.
image



Regarding the reason of parenthesis, your recommendation is changing
 "To use the NFC API, the application... " to "To use the NFC API in, the application...",
and I don't understand the reason of the change.

**Comment by mijin85cho** [comment]
@nawab-samsung Please check the green part only of the https://github.com/Samsung/tizen-docs/pull/616/files, updated point by @jh8801jung.

I recommend you to review the newly added note only, and revert the "To enable use of NFC functionality in your application:" to original sentence. We are taking to much time at where which is not the point.

File: docs/application/native/guides/connectivity/nfc.md [hunk 10]
```diff
@@ -98,9 +98,15 @@ The main features of the NFC API include:
...
+  > **Note**
+  >
+  > - HCE operations begin with the launch of HCE application in Tizen Application Framework. With the execution of these operations, the data transfer to the application needs more time.
+  > - In case, you consider time efficiency, it is recommend not to create HCE applications and communicate with the Secure Elements provided in the device.
+  > - Alternatively, ensure that the HCE application is always in the launch state.
+
 ## Prerequisites
 
-To enable your application to use the NFC functionality:
+To enable use of NFC functionality in your application:
```
Comment (nawab-samsung) [inline docs/application/native/guides/connectivity/nfc.md]: @jh8801jung
1. Change
To enable the use of NFC functionality in your application:

with 
 
To enable your application to use NFC functionality:

Reason: To maintain consistency (after discussion) 



2. Global Comment
Remove the following line just below the diagram after **Note**

Pay attention to the following:

Comment (nawab-samsung) [reply docs/application/native/guides/connectivity/nfc.md]: @mijin85cho after mutual conclusion, the above changes are still mandatory ones. 

@jh8801jung kindly do the needful ASAP

**Comment by jh8801jung** [comment]
Please check latest commit.
